zcash_client_backend/src/data_api/wallet/input_selection.rs

```diff
@@ -540,9 +556,16 @@ where
                 orchard_outputs.push(OrchardPayment(payment.amount));
             };
 
-            match &payment.recipient_address {
+            let addr_network = params.address_network().ok_or(ConversionError::User(
+                "Unable to determine network for address.",
+            ))?;
+            let recipient_address = payment
+                .recipient_address
+                .clone()
+                .convert_if_network::<Address>(addr_network)?;
```

Comment:
We shouldn't need to have to convert to a `zcash_keys::Address` at this point, but the alternative is creating a separate type just so that we can have a `TryFromAddress` impl that does the equivalent of this match.